internal/applicationsnapshot/input_test.go

```diff
@@ -223,3 +301,135 @@ func TestReadSnapshotFile(t *testing.T) {
 	})
 
 }
+
+func TestExpandImageIndex(t *testing.T) {
+	mockRemoteClient := new(MockRemoteClient)
+	mockImageIndex := new(mockIndex)
+	mockRemoteClient.
+		On("Get", mock.Anything).Return(&remote.Descriptor{Descriptor: v1.Descriptor{MediaType: "application/vnd.oci.image.index.v1+json"}}, nil)
+	mockRemoteClient.
+		On("Index", mock.Anything).Return(mockImageIndex, nil)
```

Comment:
Would be better not to allow for anything (`mock.Anything`) but assert concrete values

Reply:
Is it worth adding a todo or filing a story for later re that suggestion?

Reply:
I can add a todo for that and file an issue. Thanks!

Reply:
Added todo in a couple of places and fixed the rest